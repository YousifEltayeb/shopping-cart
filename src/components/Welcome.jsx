import styled from "styled-components";
import { Link } from "react-router";
const Welcome = () => {
  const Span = styled.span`
    color: red;
  `;
  const Links = styled.div`
    display: flex;
    gap: 1rem;
    font-size: 64px;
    justify-content: center;
  `;
  return (
    <main aria-label="Welcome">
      <h1>
        Welcome To <Span>PickFlick</Span>{" "}
      </h1>{" "}
      <p>We specilize in selling the best animation flicks</p>
      <Links>
        <Link to="/Store">Store</Link>
        <Link to="/Cart">Cart</Link>
      </Links>
    </main>
  );
};

export default Welcome;
