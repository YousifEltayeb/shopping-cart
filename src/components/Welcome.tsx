import styled from "styled-components";
const Welcome = (): React.JSX.Element => {
  const Span = styled.span`
    color: red;
  `;
  return (
    <main aria-label="Welcome">
      <h1>
        Welcome To <Span>PickFlick</Span>{" "}
      </h1>
      <p>We specilize in selling the best animation flicks</p>
    </main>
  );
};

export default Welcome;
