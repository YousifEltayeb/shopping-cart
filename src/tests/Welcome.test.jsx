import { describe, it, expect } from "vitest";
import { render } from "@testing-library/react";
import Welcome from "../components/Welcome";
import { MemoryRouter } from "react-router-dom";
describe("Welcome", () => {
    it("Welcome message is rendered", () => {
        const { container } = render(
            <MemoryRouter>
                <Welcome />
            </MemoryRouter>
        );
        expect(container).toMatchSnapshot();
    });
});
