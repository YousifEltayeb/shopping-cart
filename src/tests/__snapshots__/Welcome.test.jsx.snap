// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Welcome > Welcome message is rendered 1`] = `
<div>
  <main
    aria-label="Welcome"
  >
    <h1>
      Welcome To 
      <span
        class="sc-FEMpB dCbcyi"
      >
        PickFlick
      </span>
       
    </h1>
     
    <p>
      We specilize in selling the best animation flicks
    </p>
    <div
      class="sc-gjcoXW fcWqld"
    >
      <a
        data-discover="true"
        href="/Store"
      >
        Store
      </a>
      <a
        data-discover="true"
        href="/Cart"
      >
        Cart
      </a>
    </div>
  </main>
</div>
`;
